Automattic/wp-calypso
# Plugins: Go back to the list position after checking a plugin

Fixes https://github.com/Automattic/wp-calypso/issues/91700

## Proposed Changes

 Show the previously visited plugin on the list after when user goes back to the list of items.

We have used an approach similar to the used on Themes: https://github.com/Automattic/wp-calypso/pull/38227

## Why are these changes being made?

See https://github.com/Automattic/wp-calypso/issues/91700

## Testing Instructions

* Go to the plugin list. It can be on Plugins home (`/plugins`), categories search (ex: `/plugins/browse/ecommerce`) or term search(ex: `/plugins?s=my+plugin`)
* Scroll down
* Click on a plugin and goes to the plugin page
* Goes back to the previous list (via `Back` button from the browser or the breadcrumbs)
* Check if the clicked plugin is visible on the list

![CleanShot 2024-06-13 at 11 38 05](https://github.com/Automattic/wp-calypso/assets/5039531/f246c0d5-f369-4577-941b-a039ab80165e)


## Pre-merge Checklist
